packages/builder-vite/index.ts

```diff
@@ -6,15 +6,19 @@ import { transformIframeHtml } from './transform-iframe-html';
 import { createViteServer } from './vite-server';
 import { build as viteBuild } from './build';
 
-import type { Builder } from '@storybook/core-common';
+import type { Builder, Options } from '@storybook/core-common';
 import type { RequestHandler, Request, Response } from 'express';
-import type { UserConfig, ViteDevServer } from 'vite';
+import type { InlineConfig, UserConfig, ViteDevServer } from 'vite';
 import type { ExtendedOptions } from './types';
 
 export interface ViteStats {}
 
 export type ViteBuilder = Builder<UserConfig, ViteStats>;
 
+export type ViteFinal = {
+  viteFinal(config: InlineConfig, options: Options): Record<string, any>;
+};
```

Comment:
This type isn't quite correct. It should be a function that returns a `InlineConfig | Promise<InlineConfig>`, not a `Record<string, any>`. Also, this is a bit of a personal nit, but it's a bit awkward for the wrapper type to be named after the function it contains. 

Something like this would be more accurate. 
```typescript
export type ViteFinal = (
  config: InlineConfig,
  options: Options
) => InlineConfig | Promise<InlineConfig>;
```

Also, I have a bit of an opinionated take here: Let's just export a type that's ready to use rather than having the user import from various places and intersect on their own.

```suggestion
export type StorybookViteConfig = StorybookConfig & {
  viteFinal: (
    config: InlineConfig,
    options: Options
  ) => InlineConfig | Promise<InlineConfig>;
};
```

Reply:
Does some error occur when the StorybookConfig type referenced by Builder-Vite itself is inconsistent with the user version, Wouldn't it be better to export it in a single type? 
>  user import from various places and intersect on their own
It's very cheap to change

Reply:
Thank you for your advice!~

Reply:
Would that problem actually happen? The `@storybook/core-common` package is a peer dep, so in theory, only the user's version loads, right? Or am I misunderstanding something? Is it a transitive dep?

Reply:
oh my bad，thx for your remind